aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -63,6 +68,14 @@ impl TransactionMetadata {
                 TransactionPayload::Script(s) => (s.code().len() as u64).into(),
                 _ => NumBytes::zero(),
             },
+            entry_function_payload: match txn.payload() {
+                TransactionPayload::EntryFunction(e) => Some(e.clone()),
+                _ => None,
```

Comment:
The wildcard match is fine in this specific case. We're cloning it only if it's an entry-function payload. This will remain correct even if `TransactionPayload` is extended later.